pkg/oc/cli/admin/mustgather/mustgather.go

```diff
@@ -0,0 +1,367 @@
+		}()
+	}
+
+	// create pod
+	pod, err := o.KubeClient.CoreV1().Pods(ns.Name).Create(o.newPod(o.NodeName))
+	if err != nil {
+		return err
+	}
+
+	// wait for pod to be running (i.e. gather has completed)
+	podPhase, err := o.waitForPodNotPending(pod)
+	if err != nil {
+		return err
+	}
+	if podPhase != corev1.PodRunning {
+		return fmt.Errorf("pod is not running: %v", podPhase)
+	}
+
+	// copy the gathered files into the local destination dir
+	err = o.copyFilesFromPod(pod)
+
+	return err
+}
+
+func (o *Options) copyFilesFromPod(pod *corev1.Pod) error {
+
+	// remote tar will write to this pipe while we read from it locally
+	reader, writer := io.Pipe()
+
+	// start the remote tar
+	go func() {
+		defer writer.Close()
+		_ = (&exec.ExecOptions{
+			StreamOptions: exec.StreamOptions{
+				IOStreams: genericclioptions.IOStreams{
+					In:     nil,
+					Out:    writer,
+					ErrOut: o.Out,
+				},
+				Namespace: pod.Namespace,
+				PodName:   pod.Name,
+			},
+			Command:   []string{"tar", "cf", "-", "/must-gather"},
+			Executor:  &exec.DefaultRemoteExecutor{},
+			Config:    o.ClientConfig,
+			PodClient: o.KubeClient.CoreV1(),
+		}).Run()
+	}()
+
+	// start the untar locally
+	tarStream := tar.NewReader(reader)
+	for {
+		header, err := tarStream.Next()
+		if err != nil {
+			if err != io.EOF {
+				return err
+			}
+			break
+		}
+		localPathName := path.Join(o.DestDir, header.Name[len("must-gather/"):])
+		// make sure parent dir exists
+		localParentDir := path.Dir(localPathName)
+		if err := os.MkdirAll(localParentDir, 0755); err != nil {
+			return err
+		}
+		// handle dir
+		if header.FileInfo().IsDir() {
+			if err := os.MkdirAll(localPathName, 0755); err != nil {
+				return err
+			}
+			continue
+		}
+		// handle file
+		localFile, err := os.Create(localPathName)
+		if err != nil {
+			return err
+		}
+		defer localFile.Close()
+		if _, err := io.Copy(localFile, tarStream); err != nil {
+			return err
+		}
+		if err := localFile.Close(); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (o *Options) waitForPodNotPending(pod *corev1.Pod) (corev1.PodPhase, error) {
+	watch, err := o.KubeClient.CoreV1().Pods(pod.ObjectMeta.Namespace).Watch(metav1.ListOptions{
+		Watch:           true,
+		ResourceVersion: pod.ResourceVersion,
+		FieldSelector:   "metadata.name=" + pod.Name,
+	})
+	if err != nil {
+		return "", err
+	}
+
+	phase := pod.Status.Phase
+	for {
+		select {
+		case event, ok := <-watch.ResultChan():
+			if !ok {
+				return phase, nil
+			}
+			pod := event.Object.(*corev1.Pod)
+			phase = pod.Status.Phase
+			if phase != corev1.PodPending {
+				watch.Stop()
+			}
+		case <-time.After(10 * time.Minute):
+			watch.Stop()
+		}
+	}
+}
+
+func (o *Options) newClusterRoleBinding(ns string) *rbacv1.ClusterRoleBinding {
+	return &rbacv1.ClusterRoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "must-gather-",
+			Labels: map[string]string{
+				"app": "must-gather",
```

Comment:
namespace this label  `"oc.openshift.io/command": "must-gather"`

Reply:
Done

Reply:
I don't see this addressed?